lib/model/query/entities.js

```diff
@@ -195,9 +229,9 @@ const createEntitiesFromPendingSubmissions = (submissionEvents, parentEvent) =>
 ////////////////////////////////////////////////////////////////////////////////
 // UPDATING ENTITIES
 
-const createVersion = (dataset, entity, data, label, version, sourceId, userAgentIn = null) => ({ context, one }) => {
+const createVersion = (dataset, entity, data, label, version, sourceId, userAgentIn = null, submitterId = null) => ({ context, one }) => {
   // dataset is passed in so the audit log can use its actee id
-  const creatorId = context.auth.actor.map((actor) => actor.id).orNull();
+  const creatorId = (context.auth ? context.auth.actor.map((actor) => actor.id).orNull() : submitterId);
```

Comment:
this whole inferring the `creatorId` from the context is awkward and came from when entities were originally updated only via the PATCH api. it doesn't work with other update pathways.

Reply:
I feel like `creatorId` should be a required parameter of `createVersion()`, then the endpoint should call `createVersion()` with `context.auth.actor...`. Then `context` wouldn't need to appear here at all.

Reply:
Will do in #1013